File: 2024/AdventOfCode/Tests/Day03Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day03Test {
    private const string Example = "Inputs/03-Example.txt";
    private const string Example2 = "Inputs/03-Example2.txt";
    private const string Input = "Inputs/03.txt";
    private static Day03 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day03(Example);
            const string expected = "161";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day03(Input);
            const string expected = "184511516";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day03(Example2);
            const string expected = "48";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day03(Input);
            const string expected = "90044227";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_03_test.exs
defmodule AdventOfCode.Day03Test do
  use ExUnit.Case

  import AdventOfCode.Day03

  @day 3
  @year 2025

  @example String.trim("""
           987654321111111
           811111111111119
           234234234234278
           818181911112111
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 357
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 17074
  end

  test "part2 example" do
    result = part2(@example)

    assert result == 3_121_910_778_619
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == 169_512_729_575_727
  end
end
